compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -346,10 +361,12 @@ private NarrowedTypes getNarrowedTypesForBinaryOp(Map<BVarSymbol, NarrowedTypes>
                                                       Map<BVarSymbol, NarrowedTypes> rhsTypes, BVarSymbol symbol,
                                                       OperatorKind operator) {
         BType lhsTrueType, lhsFalseType, rhsTrueType, rhsFalseType;
+        boolean isLhsOperand = false;
```

Comment:
Does this name reflect what we're tracking? Shouldn't it be something like `usesLhsTypes`?

Reply:
Also can't we just redo the `lhsTypes.containsKey(symbol)` check in L400 instead? Purely for readability.